Source/DafnyTestGeneration/README.md

```diff
@@ -39,7 +39,7 @@ Even though it is possible to run the tests in this way, we currently provide no
 ## How to Generate Tests?
 
 - Test generation currently works with all basic types, user-defined classes, sequences, sets, and maps. It does not work with datatypes, arrays, and multisets. It is also not possible to generate tests for constructors.
-- To generate block- or path-coverage tests use the `/generateTestMode:Block` or `/generateTestMode:Path` arguments respectively. You will likely also need `definiteAssignment:3`, which will ensure that you generate tests even for those methods that do not contain any assertions or pre-\post-conditions.
+- To generate block- or path-coverage tests use the `/generateTestMode:Block` or `/generateTestMode:Path` arguments respectively. You will likely also need `definiteAssignment:3`, which forces the Dafny-to-Boogie translator to emit implementations even for those methods that don't have any assertions or pre-\post-conditions. This, in turn, allows to generate tests for such methods.
```

Comment:
This is not true. Even with `/definiteAssignment:3`, Dafny still performs the optimization of not generating a Boogie procedure if there are no proof obligations. The reason you may think the two have anything to do with each other is that `/definiteAssignment:3` causes there to be more proof obligations. Still, a method like

``` dafny
method M() {
}
```

does not generate any Boogie procedure.

I suggest you remove the mention of `/definiteAssignment:3`. Instead, you could just remark that test generation relies on Dafny to generate Boogie procedures and that Dafny does not generate a Boogie procedure when there are no proof obligations, and thus no tests will be generated for those. Alternatively, you could disable this optimization when the test-generation flags are used (see the `EmitImplementation` method in `Translator.cs`).

Reply:
Thank you for clarifying this! I have paraphrased the explanation as you suggested. 

My reasoning was that one would not generate tests for a method that doesn't take any arguments and that `definiteAssignemnt:3` adds proof obligations for all methods that do (although I might be wrong about this) but I do realize that it technically has its own purpose and that one does not necessarily want to use it. In any case, I no longer mention `definiteAssignemnt:3` in the description (although I still have to use it in the example and the lit test)